AnimView/frmMain.cpp

```diff
@@ -694,9 +731,68 @@ void frmMain::_onPanelClick(wxMouseEvent& evt) {
   }
   m_txtMoodPosition[1]->SetValue(
       wxString::Format(L"{%i, %i, \"px\"}", m_iMoodDrawX, m_iMoodDrawY));
+  printf("%ld, {%i, %i, \"px\"}, -- Anim %ld\n", m_iCurrentFrame, m_iMoodDrawX,
+         m_iMoodDrawY, m_iCurrentAnim);
+
   if (m_bDrawMood) m_panFrame->Refresh(false);
 }
 
+void frmMain::_onMoodFrameApply(wxCommandEvent& evt) {
+  wxString wxText = m_txtMoodFramePos->GetLineText(0);
+  std::string text = wxText.ToStdString();
+
+  std::regex tilePosRegex(
+      "\\{ *([-0-9]+[.][0-9]*) *, *([-0-9]+[.][0-9]*) *\\}");
+  std::regex pixelPosRegex("\\{ *([-0-9]+) *, *([-0-9]+) *, *[\"]px[\"] *\\}");
```

Comment:
`(-?\d+(?:\.\d+))?` maybe?

For a group containing a potentially negative number with an optional decimal followed by one or more digits?

Reply:
What syntax is that?  I have been looking at the c++ standard, and they have https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html#tag_09_04 

At least `(?:` isn't there.

Maybe it's simpler to write a character scanner. There is nothing complicated in there.

Reply:
https://en.cppreference.com/w/cpp/regex  is the regex cpp library

Reply:
The default c++ regex is ECMAScript - https://en.cppreference.com/w/cpp/regex/ecmascript check the section on sub-expressions. 

Specifically it's a subexpression that doesn't get captured as a match group.

Reply:
Ah ok. I know `(?:` from Python, but didn't find it in the Re documentation that I read. Should have picked a different one :)

Reply:
In the mean time, I switched to c character matching which does the job as well.